src/gwt/src/org/rstudio/studio/client/common/mirrors/model/CRANMirror.java

```diff
@@ -111,8 +111,48 @@ public final void setSecondaryRepos(ArrayList<CRANMirror> repos)
       return repos;
    }
 
+   /**
+    * Sets Name and Host as a "custom" CRAN repo defined by URL alone
+    */
+   public final void setAsCustom()
+   {
+      setName(getCustomEnumValue());
+      setHost(getCustomEnumValue());
+   }
+
+   /**
+    * Returns whether this is a "custom" CRAN repo (eg: defined by URL alone)
+    */
+   public final boolean isCustom()
+   {
+      return getHost().equals(getCustomEnumValue());
+   }
+
+   /**
+    * Returns a formatted display name for this CRANMirror.
+    *
+    * Returned name includes Name and Host if a standard host, and simply the URL of a custom host from the user
+    */
    public final String getDisplay()
    {
-      return getName() + " - " + getHost();
+      if (isCustom()) {
+         // Host is Custom with no standard Name/Host info.  Identify by URL
+         return getURL();
+      } else {
+         return getName() + " - " + getHost();
+      }
+   }
+
+   // Implement enumerators as functions because this extends a JavaScriptObject
+   // A cleaner way might be to wrap this in a wrapper class, but this gets the job done
+   public final static String getCustomEnumValue()
+   {
+      return "Custom"; //$NON-NLS-1$
```

Comment:
What do the `NON-NLS-1` comments mean?

Reply:
This is an indicator to most IDEs (at least Eclipse and IntelliJ) to ignore hard coded string literals on this line (eg: literals that should not be internationalized)